### PR TITLE
feat(Auth): Implementation of getCurrentUser api

### PR DESCRIPTION
**Pull request by royjit**
*Description of changes:*

This PR is dependent on userSub api in AWSMobileClient - https://github.com/aws-amplify/aws-sdk-ios/pull/2599/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Auth/AWSAuthPlugin/Dependency/AuthenticationProviderAdapter.swift [hunk 1]
```diff
@@ -16,13 +16,14 @@ class AuthenticationProviderAdapter: AuthenticationProviderBehavior {
         self.awsMobileClient = awsMobileClient
     }
 
-    func signInUsername() -> Result<String, AuthError> {
+    func currentUser() -> AuthUser? {
```
Comment (palpatim) [inline AmplifyPlugins/Auth/AWSAuthPlugin/Dependency/AuthenticationProviderAdapter.swift:19]: rename to `getCurrentUser`, or make this a property.

File: AmplifyPlugins/Auth/AWSAuthPlugin/Models/AWSAuthUser.swift [hunk 2]
```diff
@@ -9,8 +9,15 @@ import Amplify
 
 public struct AWSAuthUser: AuthUser {
 
+    /// The username for the logged in user
+    ///
+    /// Value maps to the username of a user in AWS Cognito User Pool. This value is set by AWS Cognito and not by the
+    /// user and does not always map with the username used to signIn.
```
Comment (palpatim) [inline AmplifyPlugins/Auth/AWSAuthPlugin/Models/AWSAuthUser.swift:15]: As noted offline, we need to figure out this behavior.

Comment (royjit) [reply AmplifyPlugins/Auth/AWSAuthPlugin/Models/AWSAuthUser.swift:15]: Keeping this for now as discussed

File: AmplifyPlugins/Auth/AWSAuthPlugin/Dependency/AuthenticationProviderAdapter.swift (hunk 1 above)
Comment (royjit) [reply AmplifyPlugins/Auth/AWSAuthPlugin/Dependency/AuthenticationProviderAdapter.swift]: Updated

File: AmplifyPlugins/Auth/AWSAuthPlugin/Service/AWSMobileClient/AWSMobileClientAdapter.swift [hunk 3]
```diff
@@ -107,6 +107,10 @@ class AWSMobileClientAdapter: AWSMobileClientBehavior {
         return awsMobileClient.username
     }
 
+    func userSub() -> String? {
```
Comment (palpatim) [inline AmplifyPlugins/Auth/AWSAuthPlugin/Service/AWSMobileClient/AWSMobileClientAdapter.swift:110]: Make this `getUserSub()` or a property

File: AmplifyPlugins/Auth/AWSAuthPlugin/Service/AWSMobileClient/AWSMobileClientBehavior.swift [hunk 4]
```diff
@@ -52,6 +52,8 @@ protocol AWSMobileClientBehavior {
 
     func username() -> String?
 
+    func userSub() -> String?
```
Comment (palpatim) [inline AmplifyPlugins/Auth/AWSAuthPlugin/Service/AWSMobileClient/AWSMobileClientBehavior.swift:55]: make this `getUserSub()` or a `userSub` property.

File: AmplifyPlugins/Auth/AWSAuthPlugin/Service/AWSMobileClient/AWSMobileClientBehavior.swift [hunk 5]
```diff
@@ -52,6 +52,8 @@ protocol AWSMobileClientBehavior {
 
     func username() -> String?
```
Comment (palpatim) [inline AmplifyPlugins/Auth/AWSAuthPlugin/Service/AWSMobileClient/AWSMobileClientBehavior.swift:53]: Make this `getUsername()` or a `username` property

**Pull request merged**